demos/basic_uq_example.py

```diff
@@ -0,0 +1,181 @@
+# this step will depend ont he specific model and what it takes as inputs. In our case
+# our model needs to know wthat the x value(s) is/are and what are the bounds on our
+# parameters
+# first we define our input range x
+xVals = np.linspace(-1*np.pi,1*np.pi,100)
+#define our parameter bounds
+#p0: amplitude
+#p1: frequency
+#p2: phase
+#p3: offset
+bounds = [0.5, 1,\
+          1, 1,\
+          1, 1,\
+          -1, 1]
+
+# next we create a model function that takes parameter values and returns the model output
+# a lambda function is used to pass in the parameter bounds, however if parameter
+# bounds are fixed then they can be hard coded into the model function and a 
+# function handle for the model can be used instead of a lambda function
+model = lambda p: modelFunction(p,x = xVals,paramBounds=bounds)
+
+# # Building the PCE
+#  First provide the indicies and distribution
+pce = PolynomialChaosExpansion(index_set, dist)
+# Next generate the samples that you want to query
+pce.generate_samples()
+print('This will query the model {0:d} times'.format(pce.samples.shape[0]))
+# Finally compute the polnomial chaos expansion of our model function
+# this next step can be run in a number of ways, the simplest is done here
+# by providing the pce with our model function. the pce will then query the model
+# function at the samples and generate the appropriate statsitcs
+pce.build(model)
+
+# The parameter samples and model evaluations are accessible:
+
+parameter_samples = pce.samples
+model_evaluations = pce.model_output
+# this means you could run the samples through a model function offline, and return
+# the outputs to the pce seperatly. See the example file ??.py for more information
+
+
+# As a sanity check: you could build a second PCE on the same parameter samples
+#  pce2 = PolynomialChaosExpansion(index_set, dist)
+#  pce2.build(model, samples=parameter_samples)
+#
+# # pce and pce2 have the same coefficients:
+#  np.linalg.norm( pce.coefficients - pce2.coefficients )
+
+# # Postprocess PCE: mean, stdev, sensitivities, quantiles
+mean = pce.mean()
+stdev = pce.stdev()
+
+# Power set of [0, 1, ..., dimension-1]
+variable_interactions = list(chain.from_iterable(combinations(range(dimension), r) for r in range(1, dimension+1)))
+
+# "Total sensitivity" is a non-partitive relative sensitivity measure per parameter.
+total_sensitivity = pce.total_sensitivity()
+
+# "Global sensitivity" is a partitive relative sensitivity measure per set of parameters.
+global_sensitivity = pce.global_sensitivity(variable_interactions)
+
+Q = 3  # Number of quantile bands to plot
+dq = 0.5/(Q+1)
+q_lower = np.arange(dq, 0.5-1e-7, dq)[::-1]
+q_upper = np.arange(0.5 + dq, 1.0-1e-7, dq)
+quantile_levels = np.append(np.concatenate((q_lower, q_upper)), 0.5)
+
+quantiles = pce.quantile(quantile_levels, M=int(2e3))
+median = quantiles[-1, :]
+
+
+# # Visualization
+M = 1000  # Generate MC samples
```

Comment:
The variable `M` doesn't seem to play a role here, meaning that you only use the first `V` samples and ignore all the rest. I'd just replace all `M` with `V` (which you set to 100)